### ISSUE TITLE
[FEAT] Have a clear way to propagate errors to forms

### ISSUE THREAD

**Issue by banuni** [pankod/refine#2131]
I want to be able to provide the form component with the server returned errors in an easy manner

**Describe the solution you'd like**
I'd imagine a `parseErrors`, `transformErrors`, `processErrors` method I could supply `useForm` that would receive the response object and would use the return value (probably an object of structure `{ fieldName: errorText[]} mapping) to set the form errors,

**Describe alternatives you've considered**
currently doing it manually using `onMutationError`, and handling the error object on my own. which doesn't allow me to use `setError` since it's a return value of `useForm`

**Comment by aliemir** [comment]
As I remember correctly, this was an issue for the notifications. I'm not sure if doing any transforms in errors are needed internally for `useForm`. I don't have any use cases other than notifications. 

If this is the case, I have a different approach in mind, let me explain:

Currently, we have `successNotification` and `errorNotification` properties in [`useForm`](https://refine.dev/docs/core/hooks/useForm/#properties). (Check types: https://refine.dev/docs/core/interfaceReferences/#successerrornotification) 

We can set those to a notification object or to `false` which disables the notification. The problem with those are, they are not aware of the response coming from the mutation. 

### Proposal

Updating the `successNotification` and `errorNotification` properties to allow functions same as `onMutationError` and `onMutationSuccess` with return type of `false` | `undefined` | `OpenNotificationParams`.

So we will have the mutation result and variables to update the notification as desired. 

`errorNotification` type will be;

```ts
interface OpenNotificationParams {
    key?: string;
    message: string;
    type: "success" | "error" | "progress";
    description?: string;
    cancelMutation?: () => void;
    undoableTimeout?: number;
}

type ErrorNotification =
    | false
    | undefined
    | OpenNotificationParams
    | ((
          error: any,
          variables: any,
          context: any,
      ) => OpenNotificationParams | false | undefined);
```

### Examples

**Same notification for all (currently supported)**

```ts
useForm({
    /* ... */
    errorNotification: {
        type: "error",
        message: "Error!",
        description: "Something went wrong"
    },
})
```

**By response**

```ts
useForm({
    /* ... */
    errorNotification: (error, variables, context) => {
        const customMessage = error?.statusCode === 500 ? "Internal Error" : "I don't know what I'm doing";
        return {
            type: "error",
            message: customMessage,
        };
    },
})
```

Do you think this will handle all the required cases? @banuni 

Waiting for comments 🚀

**Comment by salihozdemir** [comment]
It's great solution! @aliemir 

We can also change `SuccessNotification` type like your proposal. In this way, successful notifications can be opened using the data from the server.

**Comment by banuni** [comment]
Hey there :) This is pretty cool, I think it could really help. although, this isn't exactly what I'm looking for....I'm looking for a way that the errors would appear on the form itself...could it be that it's supposed to be that way but I wired something wrong?

**Comment by mayank-meragi** [comment]
@banuni I use something like this

```typescript
export const useForm = <
  TData extends BaseRecord = BaseRecord,
  TError extends HttpError = HttpError,
  TVariables = {},>(
  props: UseFormProps<TData, TError, TVariables> = {},
): UseFormReturnType<TData, TError, TVariables> => {
  const useFormProps = useFormAntd<TData, TError, TVariables>(props)
  useEffect(
    () => {
      if (useFormProps.mutationResult.isError) {
        setErrors(
          useFormProps.mutationResult.error.response.data,
          useFormProps.form,
        )
      }
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [useFormProps.mutationResult.isError],
  )

  return useFormProps
}
```


```typescript
export const setErrors = (
  errorData: { [key: string]: string[] },
  form?: FormInstance,
) => {
  Object.entries(errorData).forEach(([field, errors]) => {
    form?.setFields([{ name: field, errors: errors ?? [] }])
  })
}
```

**Comment by omeraplak** [comment]
Hey,
We've released [`@pankod/refine-core@3.46.0`](https://github.com/pankod/refine/releases/tag/%40pankod%2Frefine-core%403.46.0) 👊
